packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -219,20 +217,31 @@ export class SummaryManager implements IDisposable {
             if (shouldSummarizeState.shouldSummarize === false) {
                 this.state = SummaryManagerState.Starting;
                 summarizer.stop(shouldSummarizeState.stopReason);
-                return;
+                return "early exit2";
```

Comment:
That's an older version of my PR. I incorporated some feedback.
Let's simplify - I'll finish my PR. There is some extra feedback that I'll incorporate :)
Let's close this PR